src/main/java/fi/otavanopisto/kuntaapi/server/rest/OrganizationsApiImpl.java

```diff
@@ -723,26 +726,26 @@ public Response deleteOrganizationSetting(String organizationIdParam, String set
 
   
   /* Pages */
-
+  
   @Override
-  public Response listOrganizationPages(String organizationIdParam, String parentIdParam, String path) {
+  public Response listOrganizationPages(String organizationIdParam, String parentIdParam, String path, String search, Long firstResult, Long maxResults) {
     OrganizationId organizationId = toOrganizationId(organizationIdParam);
     if (organizationId == null) {
       return createNotFound(NOT_FOUND);
     }
     
-    boolean onlyRootPages = StringUtils.equals("ROOT", parentIdParam);
-    PageId parentId = onlyRootPages ? null : toPageId(parentIdParam);
+    if (search != null && (parentIdParam != null || path != null)) {
+      return createNotImplemented("Search parameter can not be combined with path or parentId parameters");
+    }
     
-    List<Page> result = new ArrayList<>();
+    List<Page> result = null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "result". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1854)